joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/TypeInferencePass.scala

```diff
@@ -0,0 +1,47 @@
+package io.joern.javasrc2cpg.passes
+
+import io.joern.javasrc2cpg.util.TypeInfoCalculator.TypeConstants
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.Call.PropertyNames
+import io.shiftleft.codepropertygraph.generated.nodes.Call
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import io.shiftleft.semanticcpg.language._
+import org.slf4j.LoggerFactory
+
+class TypeInferencePass(cpg: Cpg) extends ConcurrentWriterCpgPass[Call](cpg) {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  override def generateParts(): Array[Call] = {
+    cpg.call.methodFullName(s".*${TypeConstants.UnresolvedType}.*").toArray
+  }
+
+  override def runOnPart(diffGraph: DiffGraphBuilder, call: Call): Unit = {
+    val candidateMethods = cpg.method.internal.nameExact(call.name).l
```

Comment:
Please use `.nextOption` and `.hasNext` instead of creating a potentially large linked list. This will probably entail moving the parameter count check up here and losing one of the log messages, but that seems OK to me.

Reply:
Noted. I'll refactor this to avoid the explicit list conversion.